integration-tests/test_selinux.py

```diff
@@ -0,0 +1,183 @@
+        end_idx = line.find(":", start_idx)
+        if end_idx > start_idx:
+            return float(line[start_idx:end_idx])
+    except (ValueError, IndexError):
+        pass
+    return None
+
+
+def _check_denials_in_audit_log(start_time):
+    """Check for denials by reading audit.log directly."""
+    audit_log = "/var/log/audit/audit.log"
+    if not os.path.exists(audit_log):
+        return None
+
+    denial_lines = []
+    with open(audit_log, "r") as f:
+        for line in f:
+            if "insights-client" not in line or "AVC" not in line:
+                continue
+            if start_time:
+                timestamp = _extract_timestamp_from_audit_line(line)
+                if timestamp and timestamp < start_time:
+                    continue
+            denial_lines.append(line.strip())
+
+    return "\n".join(denial_lines[-10:]) if denial_lines else None
+
+
+def _check_selinux_denials(start_time=None):
+    """Return SELinux AVC denials for insights-client since start_time, or None
+    Args:
+        start_time: Unix timestamp. Only denials after this time are returned.
+                   If None, returns all available denials.
+    Returns:
+        String with denial messages or None.
+    """
+    return _check_denials_with_ausearch(start_time) or _check_denials_in_audit_log(
+        start_time
+    )
+
+
+@pytest.mark.tier2
+def test_register_unconfined_t_no_context_change(insights_client):
+    """
+    :id: 91c9c37f-b954-4fc4-84da-a4a2a9dbee9d
+    :title: Test insights-client --register runs in unconfined_t without context change
+    :description:
+        This test verifies that when running insights-client --register as root
+        in unconfined_t SELinux context with SELinux enforcing mode, the insights-core
+        code executes in unconfined_t context without any SELinux context change
+        and without any SELinux denials.
+    :tags: Tier 1
+    :reference: https://issues.redhat.com/browse/CCT-1717
+    :steps:
+        1. Save current SELinux mode
+        2. Register system using subscription-manager (already done by fixture)
+        3. Switch system to SELinux enforcing mode
+        4. Verify current context is unconfined_t
+        5. Run insights-client --register as root
+        6. Check that insights-core code was executed in unconfined_t context
+        7. Check that there was no SELinux denial during execution
+        8. Restore original SELinux mode
+    :expectedresults:
+        1. Current SELinux mode is saved
+        2. System is registered (handled by register_subman fixture)
+        3. SELinux is set to enforcing mode
+        4. Current context is confirmed as unconfined_t
+        5. insights-client --register completes successfully
+        6. insights-core runs in unconfined_t context (no transition to confined)
+        7. No SELinux denials are found in audit logs
+        8. Original SELinux mode is restored
+    """
```

Comment:
I don't think this should be as expected result. There is no assert to check that the original selinux mode is restored. So I would remove this expected result.